pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -0,0 +1,300 @@
+			return runtime.Result{}, err
+		}
+		markEvictionExecuted(&eviction, fmt.Sprintf(evictionAllowedPDBSpecifiedFmt, availableBindings, desiredBindings, totalBindings))
+	} else {
+		markEvictionNotExecuted(&eviction, fmt.Sprintf(evictionBlockedPDBSpecifiedFmt, availableBindings, desiredBindings, totalBindings))
+	}
+
+	return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+}
+
+// updateEvictionStatus updates eviction status.
+func (r *Reconciler) updateEvictionStatus(ctx context.Context, eviction *placementv1alpha1.ClusterResourcePlacementEviction) error {
+	evictionRef := klog.KObj(eviction)
+	if err := r.Client.Status().Update(ctx, eviction); err != nil {
+		klog.ErrorS(err, "Failed to update eviction status", "clusterResourcePlacementEviction", evictionRef)
+		return controller.NewUpdateIgnoreConflictError(err)
+	}
+	klog.V(2).InfoS("Updated the status of a eviction", "clusterResourcePlacementEviction", evictionRef)
+	return nil
+}
+
+// deleteClusterResourceBinding deletes the specified cluster resource binding.
+func (r *Reconciler) deleteClusterResourceBinding(ctx context.Context, binding *placementv1beta1.ClusterResourceBinding) error {
+	bindingRef := klog.KObj(binding)
+	if err := r.Client.Delete(ctx, binding); err != nil {
+		klog.ErrorS(err, "Failed to delete cluster resource binding", "clusterResourceBinding", bindingRef)
+		return controller.NewDeleteIgnoreNotFoundError(err)
+	}
+	klog.V(2).InfoS("Issued delete on cluster resource binding, eviction succeeded", "clusterResourceBinding", bindingRef)
+	return nil
+}
+
+// isEvictionAllowed calculates if eviction allowed based on available bindings and spec specified in placement disruption budget.
+func isEvictionAllowed(desiredBindings int, bindings []placementv1beta1.ClusterResourceBinding, db placementv1alpha1.ClusterResourcePlacementDisruptionBudget) (bool, int) {
+	availableBindings := 0
+	for i := range bindings {
+		availableCondition := bindings[i].GetCondition(string(placementv1beta1.ResourceBindingAvailable))
+		if condition.IsConditionStatusTrue(availableCondition, bindings[i].GetGeneration()) {
+			availableBindings++
+		}
+	}
+	var disruptionsAllowed int
+	switch {
+	case db.Spec.MaxUnavailable != nil:
+		maxUnavailable, _ := intstr.GetScaledValueFromIntOrPercent(db.Spec.MaxUnavailable, desiredBindings, true)
+		unavailableBindings := len(bindings) - availableBindings
+		disruptionsAllowed = maxUnavailable - unavailableBindings
+	case db.Spec.MinAvailable != nil:
+		minAvailable, _ := intstr.GetScaledValueFromIntOrPercent(db.Spec.MinAvailable, desiredBindings, true)
+		disruptionsAllowed = availableBindings - minAvailable
+	}
+	if disruptionsAllowed < 0 {
+		disruptionsAllowed = 0
+	}
+	return disruptionsAllowed > 0, availableBindings
+}
+
+// calculateSchedulerTargetedBindings returns the count of bindings targeted by the scheduler.
+func calculateSchedulerTargetedBindings(bindings []placementv1beta1.ClusterResourceBinding) int {
```

Comment:
This does not look right as we don't know what is the real target of a pickAll (https://kubernetes.io/docs/tasks/run-application/configure-pdb/#arbitrary-controllers-and-selectors)